code/game/objects/items/weapons/twohanded.dm

```diff
@@ -338,9 +338,167 @@
 	resistance_flags = UNACIDABLE
 	attack_speed = 12 //Default is 7.
 
-
 /obj/item/weapon/twohanded/glaive/damaged
 	name = "war glaive"
 	desc = "A huge, powerful blade on a metallic pole. Mysterious writing is carved into the weapon. This one is ancient and has suffered serious acid damage, making it near-useless."
 	force = 18
 	force_wielded = 28
+
+/obj/item/weapon/twohanded/rocketsledge
+	name = "rocket sledge"
+	desc = "Fitted with a rocket booster at the head, the rocket sledge would deliver a tremendously powerful impact, easily crushing your enemies. Uses fuel to power itself. AltClick to tighten your grip."
+	icon_state = "rocketsledge"
+	item_state = "rocketsledge"
+	force = 30
+	w_class = WEIGHT_CLASS_BULKY
+	flags_equip_slot = ITEM_SLOT_BACK
+	force_wielded = 75
+	throwforce = 50
+	throw_speed = 2
+	edge = 1
+	sharp = IS_SHARP_ITEM_BIG
+	flags_atom = CONDUCT | TWOHANDED
+	attack_verb = list("smashed", "hammered")
+	attack_speed = 20
+
+	///amount of fuel stored inside
+	var/max_fuel = 50
+	///amount of fuel used per hit
+	var/fuel_used = 5
+	///additional damage when weapon is active
+	var/additional_damage = 75
+	///stun value
+	var/stun = 1
+	///weaken value
+	var/weaken = 2
+	///stagger value
+	var/stagger = 2
+	///knockback value
+	var/knockback = 0
+
+/obj/item/weapon/twohanded/rocketsledge/Initialize()
+	. = ..()
+	create_reagents(max_fuel, null, list(/datum/reagent/fuel = max_fuel))
+
+/obj/item/weapon/twohanded/rocketsledge/equipped(mob/user, slot)
+	. = ..()
+	toggle_item_bump_attack(user, TRUE)
+
+/obj/item/weapon/twohanded/rocketsledge/dropped(mob/user)
+	. = ..()
+	toggle_item_bump_attack(user, FALSE)
+
+/obj/item/weapon/twohanded/rocketsledge/examine(mob/user)
+	..()
+	to_chat(user, "It contains [reagents.get_reagent_amount(/datum/reagent/fuel)]/[max_fuel] units of fuel!")
+
+/obj/item/weapon/twohanded/rocketsledge/wield(mob/user)
+	. = ..()
+	if((reagents.get_reagent_amount(/datum/reagent/fuel) < fuel_used))
+		playsound(loc, 'sound/items/weldingtool_off.ogg', 25)
+		return
+	update_icon()
+
+/obj/item/weapon/twohanded/rocketsledge/unwield(mob/user)
+	. = ..()
+	update_icon()
+
+/obj/item/weapon/twohanded/rocketsledge/update_icon_state()
+	if ((reagents.get_reagent_amount(/datum/reagent/fuel) > fuel_used) && (CHECK_BITFIELD(flags_item, WIELDED)))
+		icon_state = "rocketsledge_w"
+	else
+		icon_state = "rocketsledge"
+
+/obj/item/weapon/twohanded/rocketsledge/update_icon()
+	update_icon_state()
```

Comment:
you should not need that. 
![image](https://user-images.githubusercontent.com/25491240/147743213-5cbad73f-053d-4a7f-a645-f2d4fd65cb6a.png)
not sure why this exist, but this is bad. Check if you can safely remove